wrapanapi/systems/scvmm.py

```diff
@@ -0,0 +1,445 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+from __future__ import absolute_import
+
+import json
+import re
+from datetime import datetime
+from textwrap import dedent
+
+import pytz
+import tzlocal
+import winrm
+from wait_for import wait_for
+
+from wrapanapi.entities import Template, TemplateMixin, Vm, VmMixin, VmState
+from wrapanapi.exceptions import (ImageNotFoundError, VMInstanceNotFound)
+from wrapanapi.systems import System
+
+
+def convert_powershell_date(date_obj_string):
+    """
+    Converts a string representation of a Date object into datetime
+
+    PowerShell prints this as an msec timestamp
+
+    So this converts to:
+    "/Date(1449273876697)/" == datetime.datetime.fromtimestamp(1449273876697/1000.)
+    """
+    match = re.search(r'^/Date\((\d+)\)/$', date_obj_string)
+    if not match:
+        raise ValueError('Invalid date object string: {}'.format(date_obj_string))
+    return datetime.fromtimestamp(int(match.group(1)) / 1000.)
+
+
+class SCVirtualMachine(Vm):
+
+    state_map = {
+        'Running': VmState.RUNNING,
+        'PowerOff': VmState.STOPPED,
+        'Stopped': VmState.STOPPED,
+        'Paused': VmState.SUSPENDED,  # 'Paused' is scvmm's version of 'suspended'
+        'Missing': VmState.ERROR,
+        'Creation Failed': VmState.ERROR,
+    }
+
+    def __init__(self, system, raw=None, **kwargs):
+        """
+        Construct an SCVirtualMachine instance tied to a specific system
+
+        Args:
+            system: instance of SCVMMSystem
+            raw: raw json (as dict) for the VM returned by the API
+            name: name of VM
+        """
+        super(SCVirtualMachine, self).__init__(system, raw, **kwargs)
+        self._name = raw['Name'] if raw else kwargs.get('name')
+        if not self._name:
+            raise ValueError("missing required kwarg: 'name'")
+        self._run_script = self.system.run_script
+        self._get_json = self.system.get_json
+
+    @property
+    def _identifying_attrs(self):
+        return {'name': self._name}
+
+    def refresh(self):
+        """
+        Get VM from SCVMM
+
+        Force reload from host to cover cases where VM was updated directly on host
+
+        Returns:
+            raw VM json
+        """
+        script = (
+            'Get-SCVirtualMachine -Name \"{}\" -VMMServer $scvmm_server | Read-SCVirtualMachine')
+        data = self._get_json(script.format(self.name))
+        if not data:
+            raise VMInstanceNotFound(self.name)
+        self.raw = data
+        return self.raw
+
+    @property
+    def name(self):
+        return self._name
+
+    def _get_state(self):
+        self.refresh()
+        return self._api_state_to_vmstate(self.raw['StatusString'])
```

Comment:
**REQUIRED:** I think it should be
```python
return self._api_state_to_vmstate(self.raw[0]['StatusString'])
```

Reply:
`self.raw` should be a dict, not a list, not sure why this would not be working for you. A simple example I am running here:
`
In [3]: mgmt = cfme.utils.providers.get_mgmt('scvmm')

In [4]: vms = mgmt.list_vms()

In [9]: for vm in vms:
   ...:     print vm.raw['StatusString']
   ...:     
Incomplete VM Configuration
Incomplete VM Configuration
Stopped
Incomplete VM Configuration
Running
Missing
Stopped
Stopped
Stopped
Stopped
Incomplete VM Configuration
Incomplete VM Configuration
Stopped
Incomplete VM Configuration
Stopped
Stopped
Running

In [11]: for vm in vms:
    ...:     print vm._api_state_to_vmstate(vm.raw['StatusString'])
    ...:     
    ...:     
No handlers could be found for logger "wrapanapi.systems.scvmm.SCVirtualMachine"
VmState.UNKNOWN
VmState.UNKNOWN
VmState.STOPPED
VmState.UNKNOWN
VmState.RUNNING
VmState.ERROR
VmState.STOPPED
VmState.STOPPED
VmState.STOPPED
VmState.STOPPED
VmState.UNKNOWN
VmState.UNKNOWN
VmState.STOPPED
VmState.UNKNOWN
VmState.STOPPED
VmState.STOPPED
VmState.RUNNING

`